File: src/BaseHmiTypes/Screens/Base/HmiLayer.cs
namespace BaseHmiTypes.Screens.Base;

public class HmiLayer : HmiModelBase
{
    public HmiProperty<bool> Visible { get; set; } = true;

    public HmiProperty<bool> RuntimeVisible { get; set; } = true;

    public HmiProperty<bool> Locked { get; set; } = false;

    public HmiProperty<double>? VisibilityMinZoom { get; set; }

    public HmiProperty<double>? VisibilityMaxZoom { get; set; }

    public IList<HmiScreenItemBase> Items { get; } = new List<HmiScreenItemBase>();
}

File: typescript/base-hmi-types/src/screens/base/HmiLayer.ts
import { HmiModelBase } from "./HmiModelBase.js";
import { HmiProperty, staticProperty } from "./HmiProperty.js";
import { HmiScreenItemBase } from "./HmiScreenItemBase.js";

export class HmiLayer extends HmiModelBase {
  visible: HmiProperty<boolean> = staticProperty(true);
  runtimeVisible: HmiProperty<boolean> = staticProperty(true);
  locked: HmiProperty<boolean> = staticProperty(false);
  visibilityMinZoom?: HmiProperty<number>;
  visibilityMaxZoom?: HmiProperty<number>;
  readonly items: HmiScreenItemBase[] = [];
}
